graphman-client/modules/graphman.js

```diff
@@ -43,18 +43,29 @@ module.exports = {
             method: 'POST',
             rejectUnauthorized: gateway.rejectUnauthorized,
             headers: headers,
-            auth: gateway.username + ":" + gateway.password,
             body: body || {}
         };
 
+        if (gateway.username && gateway.password) {
+            req.auth = gateway.username + ":" + gateway.password;
+        }
+        else if (gateway.keyFilename && gateway.certFilename) {
+            // This expects the certificate.pem and certificate.key file(s) to be in the graphman-client directory. 
+            req.key = utils.readFileBinary(`${__dirname}/../${gateway.keyFilename}`);
+            req.cert = utils.readFileBinary(`${__dirname}/../${gateway.certFilename}`);
+        }
+        else {
+            throw new Error("Either username/password or certificate authentication must be configured. Please provide the following values: \n 'Username/Password Authentication: the username and password fields' \n 'Certificate Authentication: keyFilename and certFilename fields'.");
+        }
```

Comment:
Informative, but it is too big. There's a room for improvement. 

**For example:**
_Authentication details are missing. Please provide either basic authentication (username/password) or mTLS based authentication (keyFilename/certFilename)_

Reply:
Changed to your example